packages/vc-export/src/KiltCredentialV1.ts

```diff
@@ -0,0 +1,365 @@
+          format: 'uri',
+        },
+      },
+      required: ['id'],
+    },
+    issuer: {
+      type: 'string',
+      format: 'uri',
+    },
+    issuanceDate: {
+      type: 'string',
+      format: 'date-time',
+    },
+    credentialStatus: {
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          format: 'uri',
+        },
+        type: {
+          // public credentials may have a different revocation check, so we don't force the type here
+          type: 'string',
+        },
+      },
+      required: ['id', 'type'],
+    },
+    federatedTrustModel: {
+      type: 'array',
+      minLength: 1,
+      items: {
+        type: 'object',
+        properties: {
+          id: {
+            type: 'string',
+            format: 'uri',
+          },
+          type: {
+            type: 'string',
+          },
+        },
+      },
+    },
+    credentialSchema: {
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          format: 'uri',
+        },
+        type: {
+          type: 'string',
+          const: JSON_SCHEMA_TYPE,
+        },
+        schema: { $ref: CType.Schemas.CTypeModel.$id },
+      },
+      required: ['id', 'type'],
+    },
+    proof: {
+      type: 'object',
+      properties: {
+        type: {
+          type: 'string',
+        },
+      },
+      required: ['type'],
+    },
+  },
+  additionalProperties: false,
+  required: [
+    '@context',
+    'type',
+    'id',
+    'credentialSubject',
+    'issuer',
+    'issuanceDate',
+    'credentialStatus',
+    'credentialSchema',
+  ],
+}
+
+const schemaValidator = new JsonSchema.Validator(credentialSchema, '7')
+schemaValidator.addSchema(CType.Schemas.CTypeModel)
+
+/**
+ * Validates an object against the KiltCredentialV1 data model.
+ * Throws if object violates the [[credentialSchema]].
+ *
+ * @param credential Credential or object to be validated.
+ */
+export function validateStructure(credential: VerifiableCredential): void {
+  const { errors, valid } = schemaValidator.validate(credential)
+  if (!valid)
+    throw new CredentialMalformedError(
+      `Object not matching ${KILT_CREDENTIAL_TYPE} data model`,
+      {
+        cause: errors,
+      }
+    )
+}
+
+/**
+ * Transforms an [[ICredential]] object to conform to the KiltCredentialV1 data model.
+ *
+ * @param input An [[ICredential]] object.
+ * @param issuer The issuer of the attestation to this credential (attester).
+ * @param chainGenesisHash Genesis hash of the chain against which the credential is verifiable.
+ * @param timestamp Timestamp of the block referenced by blockHash in milliseconds since January 1, 1970, UTC (UNIX epoch).
+ * @param ctype Optional: The CType object referenced by the [[ICredential]].
+ * @returns A KiltCredentialV1 with embedded KiltAttestationProofV1 proof.
+ */
+export function fromICredential(
+  input: ICredential,
+  issuer: DidUri,
+  chainGenesisHash: Uint8Array,
```

Comment:
Should this also default to Spiritnet?

Reply:
good idea. I'd have to restructure the function signature then; but I was thinking that a destructured input object makes more sense anyway, so that you have named arguments in arbitrary order

Reply:
9a0ded49479e5153709c2e287bdb23df3ed15731